apps/integration/integration/automation/runtests.sh

```diff
@@ -33,20 +55,73 @@ checkout_main_if_new() {
 
 build_containers() {
   echo "Yep, definitely building containers. That's what this function does" >&2
+  docker-build-hub apps/crawler/crawler/http/Dockerfile
+  docker-build-hub apps/crawler/crawler/s3/Dockerfile
+  docker-build-hub apps/importer/Dockerfile.buildx
+  docker-build-hub apps/opensearch/Dockerfile
+  docker-build-hub apps/jupyter/Dockerfile.buildx --build-arg=TAG="${TAG}"
+  docker-build-hub apps/demo-ui/Dockerfile.buildx
+  docker-build-hub apps/remote-processor-service/Dockerfile.buildx
 }
 
 handle_outputs() {
   echo "Yep, definitely handling test outputs. That's what this function does" >&2
+  local passed_tests="$1"
+  mv test-output.log "${QUERY_LOGFILE}"
+  [[ ${passed_tests} = 0 ]] && touch "${RUNDIR}/passed"
+  [[ ${passed_tests} != 0 ]] && touch "${RUNDIR}/failed"
+  aws s3 cp -r "${RUNDIR}" "s3://sycamore-ci/${ARCH}"
 }
 
 runtests() {
   docker system prune -f --volumes
   docker compose up reset
-  poetry run pytest apps/integration/ -p integration.conftest --noconftest --docker-tag latest_rc
+  poetry run pytest apps/integration/ -p integration.conftest --noconftest --docker-tag "${TAG}"
   # this is a complicated command, so: ^                        ^            ^ test against containers tagged latest_rc
   #                                    |                     don't load conftest at pytest runtime; it's already loaded
   #                                     load conftest with plugins, to capture the custom command line arg --docker-tag
+  return $?
+}
+
+docker-build-hub() {
+  local docker_file="$1"
+  [[ -n "${docker_file}" ]] || error "missing ${docker_file}"
+  local repo_name="$(_docker-repo-name "${docker_file}")"
+  [[ -n "${repo_name}" ]] || error "empty repo name"
+  shift
+
+  local platform=linux/amd64,linux/arm64
+  echo
+  echo "Building in sycamore and pushing to docker hub with repo name '${repo_name}'"
+  docker buildx build "$(_docker-build-args)" -t "${repo_name}:${TAG}" -f "${docker_file}" --platform ${platform} \
+     --cache-to type=registry,ref="${repo_name}:build-cache",mode=max \
+     --cache-from type=registry,ref="${repo_name}:build-cache" \
+     "$@" --push . || error "buildx failed"
+  echo "Successfully built in $dir using docker file $docker_file"
 }
 
+_docker-repo-name() {
+  local docker_file="$1"
+  echo "Finding repo name in: ${docker_file}" >&2
+  local repo_name="$(grep '^# Repo name: ' "${docker_file}" | awk '{print $4}')"
```

Comment:
A more robust approach might be:
```
local repo_name=$(sed -nr -e 's/^#\s?Repo\s+name:\s*(\w+)/\1/ip' "${docker_file}" | head -n 1)
```
It would save you from multiple copies and variations in formatting.  It goes well with your check that the word count is one, below.  In assignment context, you don't need the quotes around `$(...)` command substitution.

Reply:
Actually, my pipe to `head -n 1` may be unnecessary if you're checking for a single word right after.

Reply:
We don't need to be very robust here. We control the input file.